### ISSUE TITLE
libjpeg8-dev is not available on Debian Jessie

### ISSUE THREAD

**Issue by atonparker** [Automattic/node-canvas#524]
Title sums it up. According to https://github.com/facebook/hhvm/issues/4616 Debian is migrating to libjpeg-turbo, but I've no idea what the difference is.

I ran into this issue when using the official [io.js:latest](https://github.com/iojs/docker-iojs/blob/d8d0b8bf6cd41da8ad4bfc63e9a7b2b098868eae/1.2/Dockerfile) docker image, as it runs on Debian Jessie. Looks like the official [node.js:latest](https://github.com/joyent/docker-node/blob/3bb9c7ba9eb2360a031717b146747eea781abfab/0.12/Dockerfile) image does as well.

**Comment by totophe** [comment]
I've tested replacing the libjpeg8-dev by the libjpeg-turbo on Debian Jessie and it seems to work so far.

```
sudo apt-get install libcairo2-dev libjpeg62-turbo-dev libpango1.0-dev libgif-dev build-essential g++
```

**Comment by dr-dimitru** [comment]
+1 for updating docs

**Comment by LinusU** [comment]
Docs should be updated...

**Comment by rugk** [comment]
In my case it did not work - even with the new module: https://github.com/Automattic/node-canvas/issues/764

**Comment by Hikariii** [comment]
Not sure how this issue is closed.
On jessie I still get errors when using `libjpeg62-turbo-dev`.
Right when I `require('canvas')` I get the following error:

```
Error: libjpeg.so.8: cannot open shared object file: No such file or directory
    at Object.Module._extensions..node (module.js:653:18)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Module.require (module.js:568:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/usr/src/app/node_modules/canvas/lib/bindings.js:3:18)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Module.require (module.js:568:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/usr/src/app/node_modules/canvas/lib/canvas.js:13:14)
```

**Comment by Hikariii** [comment]
If I fake (symlink) a libjpeg.so.8 file I also get the error:
```
Error: libgif.so.7: cannot open shared object file: No such file or directory
    at Object.Module._extensions..node (module.js:653:18)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Module.require (module.js:568:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/usr/src/app/node_modules/canvas/lib/bindings.js:3:18)
    at Module._compile (module.js:624:30)
    at Object.Module._extensions..js (module.js:635:10)
    at Module.load (module.js:545:32)
    at tryModuleLoad (module.js:508:12)
    at Function.Module._load (module.js:500:3)
    at Module.require (module.js:568:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/usr/src/app/node_modules/canvas/lib/canvas.js:13:14)
```

**Comment by Hikariii** [comment]
I am building from an official nodejs (jessie based) docker image:
```
FROM node:8

RUN apt-get update && apt-get install -y \
	libcairo2-dev \
	libjpeg-dev \
	libpango1.0-dev \
	libgif-dev \
	libpng-dev \
	build-essential \
	g++
```

**Comment by zbjornson** [comment]
@Hikariii see https://github.com/Automattic/node-canvas/pull/1019#issuecomment-337060038. (I think that PR is ready to merge.)

**Comment by Hikariii** [comment]
@zbjornson For debian libjpeg-dev does not correctly resolve to libjepeg8.
So my solution was to use a ubuntu based container.
For anyone out there trying to get this working on debian:
I used this Dockerfile to finally fix the issue with bindings.
To me it seems using debian is just broken.

```
FROM ubuntu:xenial

RUN apt-get update && apt-get install -y \
	curl \
	git

RUN curl -sL https://deb.nodesource.com/setup_8.x | bash - \
	&& curl -sL https://deb.nodesource.com/setup_8.x | bash - \
	&& curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
	&& echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list

RUN apt-get update && apt-get install -y \
	nodejs \
	yarn \
	libcairo2-dev \
	libjpeg-dev \
	libpango1.0-dev \
	libgif-dev \
	libpng-dev \
	build-essential \
	g++
```

**Comment by anthonyfok** [comment]
For the record, the removal of libjpeg8 from Debian was *not* due to migration to libjpeg-turbo.
Rather, the direct reason is that libjpeg8 is superseded by libjpeg9.
See https://bugs.debian.org/867047 for the removal request by the libjpeg8 and libjpeg9 maintainer.

The latest and final Debian libjpeg8 package may be found in the Debian snapshot archive at this link: https://snapshot.debian.org/package/libjpeg8/8d1-2/

Meanwhile, Ubuntu has kept libjpeg8 available to this day in 20.04 LTS and in groovy for maximum backward compatiblity.

**Comment by agenterp** [comment]
you can manually copy :+1: 
e.g. in Dockerfile
COPY libjpeg.so.8 /usr/lib/x86_64-linux-gnu/libjpeg.so.8
